SDKAndroid/src/main/kotlin/app/cybrid/sdkandroid/core/AssetPipe.kt

```diff
@@ -45,17 +52,29 @@ object AssetPipe {
         }
     }
 
-    private fun transformWithDecimals(value:BigDecimal, decimals: BigDecimal, unit: String) : BigDecimal {
-
-        val divisor = BigDecimal(10).pow(decimals)
-        val tradeUnit = value.div(divisor)
-        val baseUnit = value.times(divisor)
-
-        return when(unit) {
-
-            "trade" -> tradeUnit
-            "base" -> baseUnit
-            else -> BigDecimal(0)
+    /**
+     * Method to trade with buyPrice
+     * If the price is zero always returns 0
+     * The base types:
+     * crypto:
+     *      1 BTC - $n USD
+     *      input BTC - ?
+     *      (input BTC * $n USD) / 1 BTC
+     * fiat:
+     *      1 BTC - $n USD
+     *      ?     - input USD (formatted)
+     *      (input USD * 1 BTC) / $n USD
+     * **/
+    fun trade(input: BigDecimal, price: BigDecimal, base: AssetBankModel.Type, decimals: BigDecimal = BigDecimal(2)): BigDecimal {
+
+        var result = BigDecimal(0)
+        if (base == AssetBankModel.Type.crypto) {
+            result = input.times(price)
+        } else {
+            if (price != BigDecimal.zero()) {
+                result = input.divL(price).setScale(decimals.toInt())
+            }
         }
+        return result
     }
```

Comment:
Why do we need this? With trades there should always be symmetry so would expect a handler for sells as well.